### PR TITLE
Add `SendDataNotification` method

### PR DESCRIPTION
**Pull request by pedropombeiro**
This PR adds again the changes to support the PN metadata leak fix PR (status-im/status-react#7268), but this time as a renamed method.

Part of status-im/status-react#7268

### CODE REVIEW

**Review by mandrigin — APPROVED** [review]
👍

File: api/backend.go [hunk 1]
```diff
@@ -473,6 +473,18 @@ func (b *StatusBackend) NotifyUsers(message string, payload fcmlib.NotificationP
 	return err
 }
 
+// SendMessageToUsers sends a message to users, which can be used to display a push notification.
+func (b *StatusBackend) SendMessageToUsers(dataPayloadJSON string, tokens ...string) error {
```
Comment (adambabik) [inline api/backend.go]: As we expose it on `StatusBackend` maybe we can call this method less general? `SendMessageToUsers` indicates sending messages not notifications.

```suggestion
func (b *StatusBackend) SendNotifications(dataPayloadJSON string, tokens ...string) error {
```

File: lib/library.go [hunk 2]
```diff
@@ -468,6 +468,46 @@ func NotifyUsers(message, payloadJSON, tokensArray *C.char) (outCBytes *C.char)
 	return
 }
 
+// SendMessageToUsers sends messages to users represented by given tokens.
+//export SendMessageToUsers
+func SendMessageToUsers(dataPayloadJSON, tokensArray *C.char) (outCBytes *C.char) {
```
Comment (adambabik) [inline lib/library.go]: Similar issue here.

File: api/backend.go (hunk 1 above)
Comment (adambabik) [reply api/backend.go]: Also what is this `dataPayloadJSON`. Don't we want to make it a struct here? If it does not matter what's insight (a bit strange), then let's at least comment what should be in this payload.

**Comment by mandrigin** [comment]
@PombeirP do we need this method if we merged the PN upgrade? Wouldn't it be better to revert the revert of NotifyUser's change?

**Comment by pedropombeiro** [comment]
> @PombeirP do we need this method if we merged the PN upgrade? Wouldn't it be better to revert the revert of NotifyUser's change?

I asked Roman what he'd prefer and he said we should just add a second method just in case and remove the old one later, but it's true that it's much cleaner if we get rid of the cruft and just replace the method (even if we rename it).

File: api/backend.go (hunk 1 above)
Comment (pedropombeiro) [reply api/backend.go]: > SendMessageToUsers indicates sending messages not notifications.

In a sense we're indeed no longer sending notifications to users (we changed from notifications API to messaging API), but rather a generic FCM message that is captured by the RN app and then used to create a local PN. That was the reason for the name. We can rename it to `SendDataNotification` which more closely reflects what it does.

Comment (pedropombeiro) [reply api/backend.go]: > Also what is this dataPayloadJSON. Don't we want to make it a struct here? If it does not matter what's insight (a bit strange), then let's at least comment what should be in this payload.

Yeah, this is up to status-react what they want to put in here, I added a comment giving a general overview of how it looks.

Comment (pedropombeiro) [reply api/backend.go]: In fact, the only reason this method doesn't live in status-react is that the FCM library that exists for Clojure doesn't support the methods we need (at some point we can probably just fork it and create an upstream PR).

**Comment by pedropombeiro** [comment]
Build for status-react: https://ci.status.im/job/status-go/job/parallel/238/ 

<blockquote><div><strong><a href="https://ci.status.im/job/status-go/job/parallel/238/">status-go » parallel #238 [Jenkins]</a></strong></div></blockquote>

File: api/backend.go (hunk 1 above)
Comment (pedropombeiro) [reply api/backend.go]: @adambabik is the current state OK or do I need to make changes?

**Review by adambabik — APPROVED** [review]
LGTM. Thanks for the code comments!

**Pull request merged**